targets/targets.json

```diff
@@ -15,6 +15,11 @@
         "default_lib": "std",
         "bootloader_supported": false,
         "static_memory_defines": true,
+        "supported_c_lib": {
```

Comment:
The suggestion discussed was to use the toolchain name i.e. ARM, GCC_ARM, IAR.

Reply:
I just used the agreed "supported_c_lib" formats from build tools and Bartek. @evedon Any suggestion on shortening "supported_c_lib"

Reply:
> The suggestion discussed was to use the toolchain name i.e. ARM, GCC_ARM, IAR.

I do not understand the intent of this sentence. Is it to agree or disagree with the post above?
Please note that keys in the `targets.json` file are in lower case.

Reply:
> I just used the agreed "supported_c_lib" formats from build tools and Bartek. @evedon Any suggestion on shortening "supported_c_lib"

I was not talking about the key `supported_c_lib` but of the keys in that attribute (`arm_c_lib`, etc)

Reply:
I did not know that keys need to be in lower case. In that case : arm, gcc_arm, iar as you suggested.
`supported_c_lib` could have a `s` at the end but otherwise fine.

Reply:
They do not have to be lower case but the convention in the file now is to use lower case for keys. I cannot think of a reason to change that.

Reply:
What is the conclusion here, is the below supported_c_lib formats ok
        "supported_c_lib": {                                                    
                "arm": ["std"],                                           
                "gcc_arm": ["std"],                                       
                "iar": ["std"]                                            
        },

Reply:
I'd prefer "supported_c_libs" (to match "supported_toolchains")